### ISSUE TITLE
Bound operations not appended to non-contained navigation property paths with ExplicitBinding annotations

### ISSUE THREAD

**Issue by andrueastman** [microsoft/OpenAPI.NET.OData#429]
Follow up to https://github.com/microsoft/OpenAPI.NET.OData/issues/232

Related to https://github.com/microsoftgraph/msgraph-metadata/issues/449

Setting `ContainsTarget="false"` on a navigation property prevents the enabling of a function/action on nav property due to the check below preventing the generation of the path because it doesn't allow for the evaluation of the ExplicitBinding annotation.

https://github.com/microsoft/OpenAPI.NET.OData/blob/d9474cff6ae35c9a4a7ccb330ba517b64f567d2b/src/Microsoft.OpenApi.OData.Reader/Edm/ODataPathProvider.cs#L828

Given the original intention of https://github.com/microsoft/OpenAPI.NET.OData/issues/232, the generation of a path when contains `ContainsTarget="false"` should take into consideration that nav property has the explicit operation binding. 

In summary, we should have the following scenarios, 

| Scenario| ContainsTarget | ExplicitBinding present | Generate? |
|--------|--------|--------|--------|
| 1 | False | False | **NO** |
| 2 | False | True | **YES** (missing scenario) |
| 3 | True | False | **YES** (already works, as the containsTarget allows this )|
| 4 | True | True | **YES** (already works, as the containsTarget allows this)|

This blocks adding paths to nav properties where contains target is false as seen in https://github.com/microsoftgraph/msgraph-metadata/issues/449
